src/core/pubsubhub.js

```diff
@@ -12,9 +12,14 @@ export const name = "core/pubsubhub";
 
 const subscriptions = new Map();
 
+/**
+ *
+ * @param {EventTopic} topic
+ * @param  {...any} data
```

Comment:
Optional TypeScript improvement: We can make EventTopic an interface and make these functions generic over that. Then we would get proper type checking in all pubsub calls.

Reply:
Yeah, was thinking the same... but I need to refactor all the call sites... they only use a single argument.

I found a huge bug also with pubsubhub.js... it wasn't designed to handle async functions, and now we have a mess for supporting "conf.beforeSave".